descope/internal/mgmt/accesskey.go

```diff
@@ -47,14 +47,17 @@ func (a *accessKey) SearchAll(ctx context.Context, tenantIDs []string) ([]*desco
 	return unmarshalAccessKeySearchAllResponse(res)
 }
 
-func (a *accessKey) Update(ctx context.Context, id, name string, description string) (*descope.AccessKeyResponse, error) {
+func (a *accessKey) Update(ctx context.Context, id, name string, description *string) (*descope.AccessKeyResponse, error) {
 	if id == "" {
 		return nil, utils.NewInvalidArgumentError("id")
 	}
 	if name == "" {
 		return nil, utils.NewInvalidArgumentError("name")
 	}
-	body := map[string]any{"id": id, "name": name, "description": description}
+	body := map[string]any{"id": id, "name": name}
+	if description != nil {
+		body["description"] = description
```

Comment:
shouldn't it be `*description` here? since yoy want the string value of the description